actions/templates/install/index.js

```diff
@@ -0,0 +1,236 @@
+    }
+    const requiredParams = [
+      POST_PARAM_NAME
+    ];
+    errorMessages = checkMissingRequestInputs(params, requiredParams);
+    if (errorMessages) {
+      return errorResponse(400, errorMessages, logger);
+    }
+
+    // extract the user Bearer token from the Authorization header
+    const accessToken = getBearerToken(params);
+
+    try {
+      // validate the token, an exception will be thrown for a non-valid token
+      await validateAccessToken(accessToken, imsUrl, imsClientId);
+    } catch (error) {
+      return errorResponse(401, [errorMessage(ERR_RC_INVALID_IMS_ACCESS_TOKEN, error.message)], logger);
+    }
+
+    Enforcer.v3_0.Schema.defineDataTypeFormat('string', 'uuid', null);
+    Enforcer.v3_0.Schema.defineDataTypeFormat('string', 'uri', null);
+
+    // WPAR002 - skip a warning about the "allowEmptyValue" property
+    // see https://swagger.io/docs/specification/describing-parameters/ Empty-Valued and Nullable Parameters
+    // EDEV001 - skip a warning about the basepath property, needed by IO Runtime for deploying apis
+    const openapi = await Enforcer('./template-registry-api.json', { componentOptions: { exceptionSkipCodes: ['WPAR002', 'EDEV001'] } });
+
+    const body = serializeRequestBody(params);
+
+    logger.debug(`Request body: ${JSON.stringify(body)}`);
+
+    const [req, reqError] = openapi.request({
+      method: HTTP_METHOD,
+      path: '/templates/{templateId}/install',
+      body
+    });
+    if (reqError) {
+      return errorResponse(400, [errorMessage(ERR_RC_INCORRECT_REQUEST, reqError.toString().split('\n').map(line => line.trim()).join(' => '))], logger);
+    }
+
+    const template = await findTemplateById(dbParams, params.templateId);
+    if (!template) {
+      return errorResponse(404, [errorMessage(ERR_RC_INVALID_TEMPLATE_ID, `Template with id ${params.templateId} not found.`)], logger);
+    }
+    logger.debug(`Template found: ${JSON.stringify(template)}`);
+
+    // extract credentials and APIs from the template
+    const { credentials, apis } = template;
+
+    logger.debug(`Credentials found: ${JSON.stringify(credentials)}`);
+    logger.debug(`APIs found: ${JSON.stringify(apis)}`);
+
+    // credential types could be: 'oauth_server_to_server', 'apikey', 'oauthnativeapp', 'oauthwebapp', 'oauthsinglepageapp'
+    // credential flow types could be: 'adobeid', 'entp'
+    // adobeid is used for OAuth creds and entp is used for OAuthS2S and JWT creds
+
+    const mapAdobeIdCredentialTypeToPlatformType = {
+      apikey: 'apiKey',
+      oauthnativeapp: 'NativeApp',
+      oauthwebapp: 'WebApp',
+      oauthsinglepageapp: 'SinglePageApp'
+    };
+
+    const consoleClient = await consoleLib.init(accessToken, params.IMS_CLIENT_ID, 'stage');
+
+    // Console APIs only support creating one type of credential at a time
+    // Also, for current templates, we only support one credential type per template
+    // So, assuming only one credential present in credentials array
```

Comment:
did we confirm this with deepesh? only one credential type per template

Reply:
I deduced this from one of the comments from Deepesh: https://adobeio.slack.com/archives/C06TCB6EWEM/p1712591662288789?thread_ts=1712591583.542489&cid=C06TCB6EWEM
However, it'll be good to re-confirm so will ask him again.